test/persistence-hooks.suite.js

```diff
@@ -2946,6 +2946,252 @@ module.exports = function(dataSource, should, connectorCapabilities) {
 
     });
 
+    describe('PersistedModel.upsertWithWhere', function() {
+      var createdId = null;
+      var createdExtra = null;
+      var createdName = null;
+
+      it('triggers hooks in the correct order on create', function(done) {
+        monitorHookExecution();
+        TestModel.upsertWithWhere({ where: { extra: 'e0101' }},
+          { id: 'not-found', name: 'not found' },
+          function(err, record, created) {
+            if (err) return done(err);
+            createdId = record.id;
+            createdExtra = record.extra;
+            createdName = record.name;
+            hookMonitor.names.should.eql([
+              'access',
+              'before save',
+              'persist',
+              'loaded',
+              'after save',
+            ]);
+            done();
+          });
+      });
+
+      it('triggers hooks in the correct order on update', function(done) {
+        monitorHookExecution();
+        TestModel.upsertWithWhere({ where: { extra: 'e0101' }},
```

Comment:
@bajtos 
While revisiting the tests in persistence hooks , I observed that a record gets created , but  is not persisted. But this is not the case, when I run the tests in manipulation.test.js and through loopback explorer. In both the cases it works as expected and I can see the flow going through all the hooks and isNewInstance being set properly .

 Any pointers will help :
describe('PersistedModel.upsertWithWhere', function() {
      var createdId = null;
      var createdExtra = null;
      var createdName = null;

      it('triggers hooks in the correct order on create', function(done) {
        monitorHookExecution();
        TestModel.upsertWithWhere({ extra: 'e0101' },
          { id: 'empid0101', name: 'Nick Carter', extra: 'e0101' },
          function(err, record, created) {
            if (err) return done(err);
            createdId = record.id;
            createdExtra = record.extra;
            createdName = record.name;
            hookMonitor.names.should.eql([
              'access',
              'before save',
              'persist',
              'loaded',
              'after save',
            ]);
            TestModel.findById('empid0101', function(err, data) {
              if (err) return done(err);
              data.name.should.equal('Nick Carter');
              data.extra.should.equal('e0101');
              done();

            });
          });
      });
This test is successful.  Soon after that if I try the following expecting an update by this tc : 
it('triggers hooks in the correct order on update', function(done) {
        monitorHookExecution();
        TestModel.upsertWithWhere({ extra: 'e0101' },
          { name: 'Nick' },
          function(err, record, created) {
            if (err) return done(err);
            hookMonitor.names.should.eql([
              'access',
              'before save',
              'persist',
              'loaded',
              'after save',
            ]);
            TestModel.find({ extra: 'e0101' }, function(err, data) {
              if (err) return done(err);
              console.log('data is:', data);
              //data.name.should.equal('Nick');
              //data.extra.should.equal('e0101');
              done();

            });
          });

It passed the hooks order test, but when I query the model here based on the id or extra , the data appears as null.